### DISCUSSION TITLE
ich bekomme es nicht hin ;

### DISCUSSION THREAD

**Discussion by Maddin4000** [jomjol/AI-on-the-edge-device#1542] [General]
### Issue

Hallo,  ich habe mehrere ESPcam mit Modul und die Webinterface Seite sagt es Installiert aber nix läuft und blinkt (WIFI) ich bin seit 3 Tagen damit beschäftigt aber ich komme nicht weiter. Unter Arduino laufen Beispielprogramme. 

Was mach ich falsch?

Ich habe auch mit Esp Flash Tool versucht; Installation Complete aber kein Wifi.

SD Karte 4 GB Wifi ini bearbeitet aber nix blinkt das Netzwerk verbindet.

**Comment by caco3** [comment]
Was sagt die Web Console dazu?

https://jomjol.github.io/AI-on-the-edge-device/index.html

**Reply by Maddin4000** [reply]
Hallo; die stüzt ab, alles mögliche ist grün ausgegeben nur PSRAM und nochwas sind rot

**Reply by caco3** [reply]
zeigst doch bitte mal hier

**Comment by jomjol** [comment]
Schick mal einen Screenshot oder ein Serial-Logfile.

**Reply by Maddin4000** [reply]
ich kenne mich nicht so gut aus, wollte eigentlich nur meinen Gaszähler Smart machen

sagst du mir wie es geht ? gerade zeugt sie nix an

[esp-web-tools-logs (11).txt](https://github.com/jomjol/AI-on-the-edge-device/files/10200856/esp-web-tools-logs.11.txt)

**Reply by Maddin4000** [reply]
<img width="446" alt="image" src="https://user-images.githubusercontent.com/115088579/206874088-b30ecb77-7281-49b3-9626-464943e75301.png">

**Reply by caco3** [reply]
Das log zeigt folgendes:
```
E (893) psram: PSRAM ID read error: 0xffffffff
E (893) cpu_start: Failed to init external RAM!
```

D.h., das vermutlich deine Hardware defekt ist!

Suche mal in den discussions danach: https://github.com/jomjol/AI-on-the-edge-device/discussions?discussions_q=psram:%20PSRAM%20ID%20read%20error

**Reply by Maddin4000** [reply]
ich habe 5 ESPCam die können doch nicht alle kaputt sein? In Arduino Webserver Kamera fünktionieren sie alle

**Reply by caco3** [reply]
nicht kaputt, aber ev. haben sie alle kein PSRAM. Das ist zusätzlicher Arbeitsspeicher in einem externem Chip

**Reply by caco3** [reply]
vielleicht braucht die In Arduino Webserver Kamera diesen Speicher nicht, wir aber schon.

**Reply by Maddin4000** [reply]
doch ich habe irgendwo ausgelesen den Chip da stand 4mb RAM

**Reply by caco3** [reply]
wie hast du das ausgelesen?

**Reply by Maddin4000** [reply]
das weiß ich leider nicht mehr, ich versuche es gerade heraus zu finden

**Reply by caco3** [reply]
soviel ich weis, kann das `esptool` das nicht auslesen. also müsste es eine andere firmware gewesen sein.

**Reply by Maddin4000** [reply]
ja mit Powershell oder Git bash denke ich

**Reply by caco3** [reply]
wie soll das gehen?

**Reply by Maddin4000** [reply]
ich suche gerade, moment

**Reply by Maddin4000** [reply]
<img width="407" alt="image" src="https://user-images.githubusercontent.com/115088579/206875052-e110e3be-43af-4d2c-8ace-26f10a4d30ab.png">

**Reply by Maddin4000** [reply]
<img width="450" alt="image" src="https://user-images.githubusercontent.com/115088579/206875521-3fb0c23c-3c22-435d-bdd9-52a57498fc42.png">

**Reply by jomjol** [reply]
Die 4 MB die du da gelistest sind, sind der Flash-Speicher (dort liegt die Firmware). PSRAM ist ein WEITERER Speicher, der durch das ESP-Tool gar nicht ausgelesen werden kann.
Wenn du im Arduino das Standartbeispiel mit Kamera ausgewählt hast, dann funktioniert das auch ohne PSRAM. Die Auflösung der Kamera wird massiv eingeschränkt, so dass es in den kleinen Speicher passt.

**Reply by Maddin4000** [reply]
ok danke erstmal, ich habe gerade gesehen das du eine ESPCAM gefunden hast bei Ali die PSRAM hat, kannst du eine bei Amazon empfehlen?

oder andere Frage kann ich mit ESP32-Wroom-32 die Kamera betreiben?

**Reply by caco3** [reply]
> oder andere Frage kann ich mit ESP32-Wroom-32 die Kamera betreiben?

Nein, das geht nicht, das Wroom hat keinen Anschluss für die Kamera

**Reply by Maddin4000** [reply]
Ok danke dir

**Reply by Maddin4000** [reply]
![Uploading C747BAD6-2C9C-4EC8-AD9B-16D867B6891A.png…]()

**Reply by Maddin4000** [reply]
Da steht aber das der von mir gekaufte RAM hat ?

**Reply by Maddin4000** [reply]
diymore ESP32 CAM Entwicklungsplatine, WLAN/Bluetooth, ESP32 DC 5V Dual-Core-Entwicklungsplatine mit 2640 Kamera-TF-Karten-Modul https://amzn.eu/d/8GkOh57

**Reply by forrestfunk** [reply]
> diymore ESP32 CAM Entwicklungsplatine, WLAN/Bluetooth, ESP32 DC 5V Dual-Core-Entwicklungsplatine mit 2640 Kamera-TF-Karten-Modul https://amzn.eu/d/8GkOh57

Nur ein kleiner Hinweis: Ich habe exakt dieses Produkt auch bestellt und funktioniert einwandfrei bei mir.

**Reply by Maddin4000** [reply]
@forrestfunk kannst du mir helfen? Ich denke ich habe vielleicht ein Programmierfehler in Arduino mit dem Board oder so

**Reply by forrestfunk** [reply]
@Maddin4000 Tut mir leid, da kann ich dir glaube nicht wirklich helfen. Ich habe noch nie etwas mit einem Arduino gemacht. Ich habe lediglich das Ai on the edge auf den ESP geflasht etc. Ich bin eigentlich einfach nach Anleitung (zusammen mit verschiedenen Youtube videos) vorgegangen und dann hat das  ganz gut funktioniert...
